schema.go

```diff
@@ -170,7 +181,80 @@ func (schema Schema) ScanRow(row Row, model Model) error {
 			continue
 		}
 
-		return errors.Wrapf(ErrSchemaColumnRequired, "missing destination name %s in %T", column, model)
+		// sorting associations columns in case of JOIN
+		found := false
+		for key, association := range schema.associations {
+			trimed := strings.TrimPrefix(column, fmt.Sprint(association.Remote().TableName(), "."))
+			if trimed != column {
+				_, ok := associationsColumns[key]
+				if !ok {
+					associationsColumns[key] = map[string]int{}
+				}
+
+				// we keep the index in values for the scan
+				associationsColumns[key][trimed] = i
+				found = true
+			}
+		}
+
+		if found {
+			continue
+		}
+
+		rest = append(rest, column)
+	}
+
+	if len(rest) > 0 {
+		return nil, errors.Wrapf(ErrSchemaColumnRequired,
+			"missing destination name %s in %T", strings.Join(rest, ", "), model)
+	}
+
+	for key, columns := range associationsColumns {
+		// retrieve reflect field based on index
+		model := reflectx.GetReflectFieldByIndexes(value, schema.associations[key].Field.FieldIndex())
+		remote := schema.associations[key].Remote()
+
+		associationValue := reflectx.GetIndirectValue(model)
+
+		// columns concerned by the association
+		rest := make([]string, len(columns))
```

Comment:
We can define `rest := make([]string, 0, len(columns))` here so we could use `rest = append(rest, key)` instead of incrementing a counter manually. What do you think?